### PR TITLE
Enable scalafix in the scalafix build.

### PR DESCRIPTION
**Pull request by olafurpg**
I think long overdue to bootstrap and start using scalafix to develop scalafix 😄 We can start with remove unused imports and terms and then add more as we go. I found NoInfer to be too noisy, I will soon have a PR making it possible to tweak the noise by excluding certain symbols.

### CODE REVIEW

File: project/MultiScalaProject.scala [hunk 1]
```diff
@@ -126,7 +126,7 @@ object TestProject {
     MultiScalaProject(
       s"tests${sub.capitalize}",
       base(sub),
-      configure
+      configure.andThen(_.disablePlugins(scalafix.sbt.ScalafixPlugin))
```
Comment (MasseGuillaume) [inline project/MultiScalaProject.scala]: Because of https://github.com/scalacenter/scalafix/issues/512 ?

File: .scalafix.conf [hunk 2]
```diff
@@ -0,0 +1,4 @@
+rules = [
```
Comment (MasseGuillaume) [inline .scalafix.conf]: explicit return type for public methods?

**Comment by MasseGuillaume** [comment]
It would be nice to add a --test in the ci matrix.

**Comment by olafurpg** [comment]
We run --test in the ci-fast-212 entry @MasseGuillaume https://travis-ci.org/scalacenter/scalafix/jobs/319146392#L942

**Pull request merged**